source/sema.h

```diff
@@ -700,13 +1088,7 @@ class sema
     ) const
         -> bool
     {
-        //  If this is a member variable in a constructor, the name doesn't
-        //  appear lexically right in the constructor, so prepending "this."
-        //  to the printed name might make the error more readable to the programmer
         auto name = decl->identifier->to_string();
-        if (decl->declaration->parent_is_type()) {
-            name += " (aka this." + name + ")";
-        }
```

Comment:
I checked the error messages this function generates.
None apply in `operator=`.
Except if you were to implement
> the init-before-use rather than init-first semantics
> -- <https://github.com/hsutter/cppfront/issues/475#issuecomment-1806144586>

in which case we could defer to this `sema.h` checks
rather than the special handling done for `operator=` in `to_cpp1.h`.

Reply:
Aha, that's right! Because I initially started implementing init-before-use semantics, and then when I switched to init-first semantics I left that init-before-use code alone with the idea of switching back in the future.